replica/distributed_loader.cc

```diff
@@ -214,7 +214,7 @@ distributed_loader::process_upload_dir(distributed<replica::database>& db, shard
                 [] (const sstables::shared_sstable&) { return true; }).get();
 
         // Move to staging directory to avoid clashes with future uploads. Unique generation number ensures no collisions.
-        const bool use_view_update_path = db::view::check_needs_view_update_path(vb.local(), erm->get_token_metadata(), *global_table, streaming::stream_reason::repair).get();
+        const bool use_view_update_path = db::view::check_needs_view_update_path(vb.local(), erm->get_token_metadata_ptr(), *global_table, streaming::stream_reason::repair).get();
```

Comment:
This caller is safe, since it holds erm.

Reply:
Correct, but I'd rather change the signature to discourage passing a token_metadata& to an async function

Reply:
Why? If the caller awaits we avoid the reference-counting overhead.

It doesn't matter much here.